File: src/main/java/alfio/controller/api/v2/model/EventCode.java
/**
 * This file is part of alf.io.
 *
 * alf.io is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * alf.io is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with alf.io.  If not, see <http://www.gnu.org/licenses/>.
 */
package alfio.controller.api.v2.model;

import alfio.model.PromoCodeDiscount;
import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public class EventCode {

    private final String code;
    private final EventCodeType type;
    private final PromoCodeDiscount.DiscountType discountType;
    private final String discountAmount;

    public enum EventCodeType {
        @Deprecated SPECIAL_PRICE, DISCOUNT, ACCESS
    }
}

File: frontend/projects/public/src/app/model/event-code.ts
export type EventCode = {
    code: string;
    type: EventCodeType;
    discountType: DiscountType;
    discountAmount: string;
}

export type EventCodeType = 'SPECIAL_PRICE' | 'DISCOUNT' | 'ACCESS';
export type DiscountType = 'FIXED_AMOUNT' | 'FIXED_AMOUNT_RESERVATION' | 'PERCENTAGE' | 'NONE';

export type DynamicDiscount = {
    discount: string;
    discountType: DiscountType;
    formattedMessage: {[key: string]: string};
}
